platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,309 @@
+            String configFile) {
+        String solrUrl = StringUtils.defaultIfBlank(url, SystemBaseUrl.constructUrl("/solr"));
+        String coreUrl = url + "/" + coreName;
+
+        RetryPolicy retryPolicy = new RetryPolicy()
+                .withBackoff(10, TimeUnit.MINUTES.toMillis(1), TimeUnit.MILLISECONDS);
+        return Failsafe.with(retryPolicy)
+                .with(executor)
+                .onRetry((c, failure, ctx) -> LOGGER.debug(
+                        "Attempt {} failed to create HTTP Solr client ({}). Retrying again.",
+                        ctx.getExecutions(),
+                        coreName))
+                .onFailedAttempt(failure -> LOGGER.debug(
+                        "Attempt failed to create HTTP Solr client (" + coreName + ")",
+                        failure))
+                .onSuccess(client -> LOGGER.debug("Successfully created HTTP Solr client ({})",
+                        coreName))
+                .onFailure(failure -> LOGGER.warn(
+                        "All attempts failed to create HTTP Solr client (" + coreName + ")",
+                        failure))
+                .get(() -> createSolrHttpClient(solrUrl, coreName, configFile, coreUrl));
+    }
+
+    private static SolrClient createSolrHttpClient(String url, String coreName, String configFile,
+            String coreUrl) throws IOException, SolrServerException {
+        SolrClient client;
+        if (StringUtils.startsWith(url, "https")) {
+            createSolrCore(url, coreName, configFile, getHttpClient(false));
+            client = new HttpSolrClient(coreUrl, getHttpClient(true));
+        } else {
+            createSolrCore(url, coreName, configFile, null);
+            client = new HttpSolrClient(coreUrl);
+        }
+        return client;
+    }
+
+    private static CloseableHttpClient getHttpClient(boolean retryRequestsOnError) {
+        SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(
+                getSslContext(), getProtocols(), getCipherSuites(),
+                SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER);
+        HttpRequestRetryHandler solrRetryHandler = new SolrHttpRequestRetryHandler();
+
+        HttpClientBuilder builder = HttpClients.custom()
+                .setSSLSocketFactory(sslConnectionSocketFactory)
+                .setDefaultCookieStore(new BasicCookieStore())
+                .setMaxConnTotal(128)
+                .setMaxConnPerRoute(32);
+
+        if (retryRequestsOnError) {
+            builder.setRetryHandler(solrRetryHandler);
+        }
+
+        return builder.build();
+    }
+
+    private static String[] getProtocols() {
+        if (System.getProperty("https.protocols") != null) {
+            return StringUtils.split(System.getProperty("https.protocols"), ",");
+        } else {
+            return DEFAULT_PROTOCOLS.toArray(new String[DEFAULT_PROTOCOLS.size()]);
```

Comment:
Isn't DEFAULT_PROTOCOLS derived from the same system property (https.protocols)?